test/integration/action_execution.go

```diff
@@ -314,6 +334,68 @@ func depSingleContainerWithResources(namespace, claimName string, replicas int32
 	return &dep
 }
 
+func createBarePod(client *kubeclientset.Clientset, pod *corev1.Pod) (*corev1.Pod, error) {
+	var errInternal error
+	var barepod *corev1.Pod
+	if err := wait.PollImmediate(framework.PollInterval, framework.TestContext.SingleCallTimeout, func() (bool, error) {
+		barepod, errInternal = client.CoreV1().Pods(pod.Namespace).Create(context.TODO(), pod, metav1.CreateOptions{})
+		if errInternal != nil {
+			glog.Errorf("Unexpected error while creating pod: %v", errInternal)
+			return false, errInternal
+		}
+		return true, nil
+	}); err != nil {
+		return nil, err
+	}
+
+	return waitForBarePod(client, barepod.Name, barepod.Namespace)
+}
+
+func genBarePodWithResources(namespace, claimName string, replicas int32, withVolume bool) *corev1.Pod {
```

Comment:
You could refactor portions of `genBarePodWithResources ` , `depSingleContainerWithResources`, etc.  There is overlap.

Reply:
Resolving since this is covered by the deploymentconfig pr